lib/mix/tasks/compile/surface.ex

```diff
@@ -201,6 +201,10 @@ defmodule Mix.Tasks.Compile.Surface do
     end
   end
 
+  defp print_diagnostic(message, :warning, file, {line, _column}) do
+    print_diagnostic(message, :warning, file, line)
+  end
+
```

Comment:
Hey @mayel! Thanks for the PR. 

I wonder if could have something like this, instead:

```elixir
if Version.match?(System.version(), ">= 1.14.0") do    
  defp print_diagnostic(message, :warning, file, {line, col}) do
    IO.warn(message, file: file, line: line, column: col)
  end
end

# TODO: Remove this clause in Surface v0.13 and set required elixir to >= v1.14
defp print_diagnostic(message, :warning, file, line) do
  rel_file = file |> Path.relative_to_cwd() |> to_charlist()
  IO.warn(message, [{nil, :__FILE__, 1, [file: rel_file, line: line]}])
end
```

This way most users could already benefit from the column info and have better diagnostics. 

@tiagoefmoraes WDYT?

Reply:
Best of both worlds!